### ISSUE TITLE
"dynamic" repository with external links to repositories

### ISSUE THREAD

**Issue by dmnc-net** [dmnc-net/repository.dmnc#2]
# Goal 
This "repo" has been created with aim to simplify the Kodi deployment. The 1st goal was make dynamic repository of 3rd party repositories as like as symlinks in your fs.

Why dynamic (working name)?
### a) static ZIP (offline)
- addon.xml (zipped) contants list of 3rd party repos (external links their official URLs)
If some repository has changed, died or I need to add new then I have to modify XML, create new ZIP and upload to all Kodi installations - **unwanted**

With help of @vlmaksime I've managed to create static repo/addon which works and contains other repositories without hosting their ZIPs.

### b) dynamic
- zip installs repo hosted on http server, repo contains list of 3rd party repos (external links their official URLs)
Repository ZIP would be installed during Kodi deployment and Kodi will fetch fresh data
... in case of changes I'll just modify XML on http server and every Kodi installation will fetch fresh data 

I found out that Kodi tries to download 3rd party repository ZIP from my repo data dir ... discussed [here](https://forum.kodi.tv/showthread.php?tid=353126)
```
ERROR: CCurlFile::Open failed with code 404 for http://localhost/repository.dmnc/repository.castagnait/repository.castagnait-1.0.1.zip:
```
As far as I found the error is produced by calling [CAddonInstallJob::DoWork](https://github.com/xbmc/xbmc/blob/6de683c281df336a844addf15bddbaf82ed3daec/xbmc/addons/AddonInstaller.cpp#L488).

At this moment reading the source code is not easy for me and documentation is not very useful in this case:
- https://kodi.wiki/view/Addon.xml
- https://kodi.wiki/view/Add-on_repositories

**Comment by vlmaksime** [comment]
> I found out that Kodi tries to download 3rd party repository ZIP from my repo data dir ... discussed here

I installed your repository and there is no such error.
Did you delete old repository from Kodi?

**Comment by dmnc-net** [comment]
Yes, this is working very well as you've recommended to me. Problem is that I need to achieve original goal. This is "static", if I'm going to release a new version (i.e. add another sub-repo to my AIO), then I have to deploy new ZIP and you (or any user) has to manually install the ZIP again.

I want to do a ZIP step once and provide this AIO repo of repos here on github. But I'm afraid that is not possible. And this "dynamic" approach is not here on github, I'm testing it just offline (you know better tune something on local http rather than commit every change).

**Comment by vlmaksime** [comment]
You can create addons.xml for your repository and include it into your dynamic repo.
Here is example with my repository:
- [repository.xml](https://github.com/vlmaksime/repository.vlmaksime/blob/gh-pages/repo/helix/repository.xml) repositiry for Kodi 14-18 repository
- [addon.xml](https://github.com/vlmaksime/repository.vlmaksime/blob/master/helix/repository.vlmaksime/addon.xml) - content of my repository addon

When I need to update the repository, I edit the contents of the repository.xml
